src/main/java/edit_public_profile_usecase/editPublicProfileInteractor.java

```diff
@@ -0,0 +1,56 @@
+package edit_public_profile_usecase;
+
+import Entities.User;
+
+public class editPublicProfileInteractor implements editPublicProfileInputBoundary{
+    final editPublicProfileDSGateway profileDSGateway;
+    final editPublicProfilePresenter profilePresenter;
+    final String username;
+
+    public editPublicProfileInteractor(String username, editPublicProfileDSGateway profileDSGateway, editPublicProfilePresenter profilePresenter) {
+        this.username = username;
+        this.profileDSGateway = profileDSGateway;
+        this.profilePresenter = profilePresenter;
+    }
+
+    @Override
+    public editPublicProfileResponseModel saveEdits(editPublicProfileRequestModel requestModel) {
+        /*Creating a failed response model*/
+        editPublicProfileResponseModel profileFailedResponseModel = new editPublicProfileResponseModel(
+                requestModel.getPreferences(),
+                requestModel.getBio(), "");
+
+        /*Returning a failed view when preferences is left blank*/
+        for (String p: requestModel.getPreferences().values()) {
+            if (p.equals("")) {
+                profileFailedResponseModel.setMessage("Preferences is left blank.");
+                return profilePresenter.prepareFailView(profileFailedResponseModel);
+            }
+        }
+
+        /*Returning a failed view when profile bio is left blank*/
+        if (requestModel.getBio().equals("")) {
+            profileFailedResponseModel.setMessage("Biography was left blank.");
+            return profilePresenter.prepareFailView(profileFailedResponseModel);
+        }
+
+        /*Set user's profile and saving user*/
+        if (profileDSGateway.existsByUsername(this.username)) {
+            User user = profileDSGateway.findUser(this.username);
+            user.getPublicProfile().setPreferences(requestModel.getPreferences());
+            user.getPublicProfile().setBiography(requestModel.getBio());
+            //TODO: save user in repo
```

Comment:
Nice job covering so many corner cases! I think all you have to do to save the user is call the UserDSGateway!